pkg/abap/build/bfw.go

```diff
@@ -0,0 +1,374 @@
+	b.Startedby = jBuild.Build.Startedby
+	b.StartedAt = jBuild.Build.StartedAt
+	b.FinishedAt = jBuild.Build.FinishedAt
+	return nil
+}
+
+func (b *Build) getTasks() error {
+	if len(b.Tasks) == 0 {
+		appendum := "/builds('" + b.BuildID + "')/tasks"
+		body, err := b.Connector.Get(appendum)
+		if err != nil {
+			return err
+		}
+		var jTasks jsonTasks
+		json.Unmarshal(body, &jTasks)
+		b.Tasks = jTasks.ResultTasks.Tasks
+		sort.Slice(b.Tasks, func(i, j int) bool {
+			return b.Tasks[i].TaskID < b.Tasks[j].TaskID
+		})
+		for i := range b.Tasks {
+			b.Tasks[i].connector = b.Connector
+		}
+	}
+	return nil
+}
+
+func (b *Build) getValues() error {
+	if len(b.Values) == 0 {
+		appendum := "/builds('" + b.BuildID + "')/values"
+		body, err := b.Connector.Get(appendum)
+		if err != nil {
+			return err
+		}
+		var jValues jsonValues
+		json.Unmarshal(body, &jValues)
+		b.Values = jValues.ResultValues.Values
+		for i := range b.Values {
+			b.Values[i].connector = b.Connector
+		}
+	}
+	return nil
+}
+
+func (b *Build) getLogs() error {
+	if err := b.getTasks(); err != nil {
+		return err
+	}
+	for i := range b.Tasks {
+		if err := b.Tasks[i].getLogs(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// PrintLogs : Returns the Build logs
+func (b *Build) PrintLogs() error {
+	if err := b.getTasks(); err != nil {
+		return err
+	}
+	for i := range b.Tasks {
+		if err := b.Tasks[i].printLogs(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (b *Build) getResults() error {
+	if err := b.getTasks(); err != nil {
+		return err
+	}
+	for i := range b.Tasks {
+		if err := b.Tasks[i].getResults(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (t *task) printLogs() error {
+	if err := t.getLogs(); err != nil {
+		return err
+	}
+	for _, logs := range t.Logs {
+		logs.print()
+	}
+	return nil
+}
+
+// GetResult : Returns the last Build artefact created from build step
+func (b *Build) GetResult(name string) (Result, error) {
+	var Results []Result
+	var returnResult Result
+	if err := b.getResults(); err != nil {
+		return returnResult, err
+	}
+	for _, task := range b.Tasks {
+		for _, result := range task.Results {
+			if result.Name == name {
+				Results = append(Results, result)
+			}
+		}
+	}
+	switch len(Results) {
+	case 0:
+		return returnResult, errors.New("No result named " + name + " was found")
+	case 1:
+		return Results[0], nil
+	default:
+		return returnResult, errors.New("More than one result with the name " + name + " was found")
+	}
+}
+
+// IsFinished : Returns Build run state
+func (b *Build) IsFinished() bool {
+	if b.RunState == Finished || b.RunState == Failed {
+		return true
+	}
+	return false
+}
+
+func (t *task) getLogs() error {
+	if len(t.Logs) == 0 {
+		appendum := fmt.Sprint("/tasks(build_id='", t.BuildID, "',task_id=", t.TaskID, ")/logs")
+		body, err := t.connector.Get(appendum)
+		if err != nil {
+			return err
+		}
+		var jLogs jsonLogs
+		json.Unmarshal(body, &jLogs)
```

Comment:
In my experience, some error handling would be good for Unmarshal calls.

Reply:
Hi Daniel,

good point. As you've might seen I've more or less only moved coding and adjusted only small parts. Will add an error handling for the unmarshal calls.

Best regards
Christian